### DISCUSSION TITLE
Does this plug-in support ssl?

### DISCUSSION THREAD

**Discussion by HG9558** [ra1u/redis-dart#90] [Q&A]
I want to use ssl to connect, is it supported?

**Comment by ra1u** [comment]
Yes.
https://github.com/ra1u/redis-dart#tls

**Comment by HG9558** [comment]
是要按着这种固定的格式来写么？
<img width="941" alt="Snipaste_2023-11-03_15-55-33" src="https://github.com/ra1u/redis-dart/assets/22909991/918b8ca2-6419-4f3d-9b4e-7c5d902d5390">

**Comment by HG9558** [comment]
I did this but couldn't connect

**Comment by ra1u** [comment]
What we have in this example are two things.

- Openning connection over secure tls protocol
- Authenticating on server with username/password

If you can not connect on server, make sure, that your address/port and cerificates on server are valid. 
I suggest that you first try to connect on server using command `redis-cli --tls`

If you use custom tls certificates, you wont be able to connect using [`RedisConnection.connectSecure`](https://github.com/ra1u/redis-dart/blob/23af8c6c7be642043995879bf3e3ebbec2aae0f5/lib/connection.dart#L30) because by defult it expects that certificates are valid.

For this reason we have [`RedisConnection.connectWithSocket`](https://github.com/ra1u/redis-dart/blob/23af8c6c7be642043995879bf3e3ebbec2aae0f5/lib/connection.dart#L40). Here you can [`connect`](https://api.flutter.dev/flutter/dart-io/SecureSocket/connect.html) with socket from dart libraray and use connected socket, to establish protocol over redis.

**Comment by HG9558** [comment]
OK, thanks
